pkg/k8s/identitybackend/identity.go

```diff
@@ -308,8 +315,19 @@ func (c *crdBackend) Release(ctx context.Context, id idpool.ID, key allocator.Al
 	return nil
 }
 
+func GetIdentitiesByKeyFunc(keyType allocator.AllocatorKey) func(obj interface{}) ([]string, error) {
+	return func(obj interface{}) ([]string, error) {
+		if identity, ok := obj.(*v2.CiliumIdentity); ok {
+			return []string{keyType.PutKeyFromMap(identity.SecurityLabels).GetKey()}, nil
+		}
+		return []string{}, nil
```

Comment:
Seems like we should return an error here.

Reply:
Fixed.